src/cljs/main/broadfcui/common/components.cljs

```diff
@@ -186,61 +186,75 @@
   {:get-fields
    (fn [{:keys [refs]}]
      {"methodVersion" (int (common/get-text refs "snapshotId"))})
+   :clear-redacted-snapshot
+   (fn [{:keys [state]}]
+     (swap! state dissoc :redacted-snapshot))
+   :get-initial-state
+   (fn [{:keys [props]}]
+     (when (:redacted? props) {:redacted-snapshot (get-in props [:entity :snapshotId])}))
    :render
    (fn [{:keys [props state this]}]
      [:div {} (when-let [wdl-parse-error (:wdl-parse-error props)] (style/create-server-error-message wdl-parse-error))
-      (let [entity (:entity props)
+      (let [{:keys [entity redacted?]} props
             config? (contains? entity :method)]
         [:div {:style {:backgroundColor (:background-light style/colors)
                        :borderRadius 8 :border style/standard-line
                        :padding "1rem"}}
          (this :render-details entity)
-         [:div {:style {:paddingTop "0.5rem"}}
-          [:span {:style {:fontWeight 500 :marginRight "1rem"}} (if config? "Referenced Method:" "WDL:")]
-          (links/create-internal {:onClick #(swap! state update :payload-expanded not)}
-                                 (if (:payload-expanded @state) "Collapse" "Expand"))]
-         (when (:payload-expanded @state)
+         (when-not redacted?
+           [:div {:style {:paddingTop "0.5rem"}}
+            [:span {:style {:fontWeight 500 :marginRight "1rem"}} (if config? "Referenced Method:" "WDL:")]
+            (links/create-internal {:onClick #(swap! state update :payload-expanded not)}
+                                   (if (:payload-expanded @state) "Collapse" "Expand"))])
+         (when (and (not redacted?) (:payload-expanded @state))
```

Comment:
why not pull this into the preceding `when-not` and just make it `when (:payload-expanded @state)`?

Reply:
i did this before but was not getting all the expected display elements because only the last element is returned from when.